Fixes to make the project work with OpenFST 1.8.3

danijel3 opened the pull request:
Most of the world has moved to version 1.8.3 of OpenFST. I made some cosmetic changes to allow the project to compile correctly.

Note, there are still a ton of warnings during compilation. I haven't made thorough checks to see if everything works (it seems to), but it does compile everything.

danijel3 committed:
Merge pull request #1 from AdolfVonKleist/master

Merge pull request #65 from danijel3/master

danijel3 committed:
Merge branch 'AdolfVonKleist:master' into master

Commit:
Fixed bugs to make compatible with openfst-1.8.3

Commit:
Changed C++ std to 17 because openfst 1.8.3 doesn't compile on c++0x

eginhard commented:
I do have a [fork](https://github.com/eginhard/phonetisaurus) where I merged some of the fixes made in various PRs here. I used it to publish a [PyPI package](https://pypi.org/project/phonetisaurus-bindings) with the prebuilt Python bindings and all necessary OpenFST binaries. I could update that to OpenFST 1.8.3

danijel3 commented:
The reason I did this is because I used Phonetisaurus extensively with Kaldi and they moved on to 1.8.3. I do recommend the update for standards adherence, but it feels to me that there are probably more changes worthwhile than the couple I made here. Oth, I've been using it in this state for a few hours and so far so good...

danijel3 commented:
Thanks for the fork. If you need help with anything just let me know.

Pallas13 commented:
Have some update this? Phonetisauru worked? I go develop a TTS Engine for Android using Flite, I go switch g2p because trained of g2p (Random Forest) of Flite is very slower. Because is trained using 1 core of CPU.

Commit:
fix std namespace

Commit:
fix dockerfile

Commit:
updated readme

Commit:
Update utf8 library

Commit:
Update codes

Commit:
Update code

Commit:
Fix null results

Commit:
Fix utf8 library not found

Commit:
Restore add -funroll-loops
CXXFLAGS "-ffast-math" in phonetisaurus-g2pfst not generate pronunciation for all word

danijel3 committed:
Merge pull request #2 from NietteLabs/master

Update utfcpp library and fix error of nil pronunciations for all words

Pallas13 committed:
Support for KenLM

Add support for use KenLM for train ngram model

Pallas13 committed:
Fix arg for train ngram model using KenLM

Fix args for train ngram model using kenLM.

danijel3 committed:
Merge pull request #3 from NietteLabs/kenlm

Support for KenLM
